File: python/examples/node_probe.py
"""Point the client at any base address and ask which kind of target it is — a coordinator or a
solo node — with one round trip, then use the node-only routes if it turns out to be a node.

INFERHUB_BASE=http://localhost:5080/ INFERHUB_API_KEY=... python examples/node_probe.py
"""

import os

from inferhub_client import InferHubClient

base_url = os.environ.get("INFERHUB_BASE", "http://localhost:5080/")
api_key = os.environ.get("INFERHUB_API_KEY")

with InferHubClient(base_url, api_key) as client:
    probe = client.probe()
    print("target kind:", probe.kind, "version:", probe.version)

    if probe.kind == "hub":
        print("nodes on the fleet:", len(probe.hub_status.nodes or []))
    else:
        print("node name:", probe.node_status.name)
        print(
            "backend:",
            probe.node_status.backend.name if probe.node_status.backend else None,
        )
        # Node-only routes: /api/collections, not the admin-gated /api/admin/vector/collections.
        for collection in client.list_node_collections():
            print(f"  collection {collection.name!r}: dimension={collection.dimension}")

File: js/examples/node-probe.ts
/**
 * probe() tells a coordinator apart from a solo node with one GET /api/status — same address,
 * same client, discriminated on whether the body carries `mode`.
 *
 * INFERHUB_BASE=http://localhost:5080/ INFERHUB_API_KEY=... npx tsx examples/node-probe.ts
 */

import { InferHubClient } from "../src/index.js";

const baseUrl = process.env.INFERHUB_BASE ?? "http://localhost:5080/";
const apiKey = process.env.INFERHUB_API_KEY;

const client = new InferHubClient({ baseUrl, apiKey });

const result = await client.probe();

if (result.kind === "hub") {
  console.log(`hub ${result.version}, ${result.hubStatus?.nodes?.length ?? 0} node(s)`);
} else {
  console.log(
    `solo node ${result.version} (${result.nodeStatus?.name}), ` +
      `capabilities: ${result.nodeStatus?.capabilities.join(", ")}`,
  );
}
